controlplane/kubeadm/controllers/controller_test.go

```diff
@@ -164,8 +162,9 @@ func TestReconcileUpdateObservedGeneration(t *testing.T) {
 	g.Expect(testEnv.Create(ctx, kcp)).To(Succeed())
 
 	// read kcp.Generation after create
-	errGettingObject := testEnv.Get(ctx, util.ObjectKey(kcp), kcp)
-	g.Expect(errGettingObject).NotTo(HaveOccurred())
```

Comment:
This needs to be in an Eventually because the cache might take a while to get updated